cardea/problem_definition/readmission.py

```diff
@@ -0,0 +1,144 @@
+
+            cutoff_times = es[self.cutoff_entity].df[self.cutoff_time_label]
+            cutoff_times = cutoff_times.to_frame()
+            cutoff_times.index = es[self.cutoff_entity].df.index
+            cutoff_times = cutoff_times.rename(columns={self.cutoff_time_label:
+                                                        'cutoff_time'})
+
+            return (es, self.target_entity,
+                    updated_es[self.target_entity].df[self.target_label],
+                    cutoff_times)
+        except ValueError:
+            raise ValueError('Cutoff time label {} in table {} does not exist'
+                             .format(self.cutoff_time_label, self.target_entity))
+
+    def generate_target_label(self, es):
+        """Generates target labels in the case of having missing label in the entityset.
+
+            Args:
+            es: fhir entityset.
+            target_label: The target label of the prediction problem.
+            target_entity: The entity name in which it contains the target label.
+
+            Returns:
+            Updated entityset with the generated label.
+
+            Raises:
+            ValueError: An error occurs if the target label cannot be generated.
+            """
+        generate_from = 'Period'
+        end = 'end'
+
+        try:
+            if self.check_target_label(
+                es,
+                generate_from,
+                end) and self.check_target_label(es,
+                                                 self.target_entity,
+                                                 'period'):
+
+                try:
+                    self.check_target_label_values(
+                        es,
+                        generate_from,
+                        end)
+
+                    entity_set_df = es[self.target_entity].df
+                    generate_df = es[generate_from].df
+                    merged_df = pd.merge(entity_set_df, generate_df,
+                                         left_on='period', right_on='object_id')
+
+                    generated_target_label = []
+                    encounter_identifier = []
+
+                    for patient in set(merged_df['subject']):
+                        patient_visits = merged_df[merged_df['subject'] == patient]
+                        inital_date = patient_visits[end].iloc[0]
+
+                        encounter_identifier.append(patient_visits['identifier'].iloc[0])
+                        generated_target_label.append(False)  # first visit
+
+                        if len(patient_visits) != 1:
+                            for visit_date, encounter_id in zip(patient_visits[end][1:],
+                                                                patient_visits['identifier'][1:]):
+
+                                visit_range = visit_date - inital_date
+                                inital_date = visit_date
+
+                                if visit_range.days <= 30:
```

Comment:
@ShahadThobaiti please make the 30 day assumption a variable that can be overridden
```suggestion
                                if visit_range.days <= self.readmission_threshold:
```